File: c++/338_Counting_Bits.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
	vector<int> countBits(int num) {
		vector<int> ans;
		ans.push_back(0);
		for (int i = 1; i < num + 1; i++){
			ans.push_back(ans.at(i >> 1) + (i & 1));
		}
		return ans;
	}
};

int main(){
	Solution s;
	int num;
	cout << "Input the int: ";
	cin >> num;
	vector<int> temp = s.countBits(num);
	cout << "[";
	for (int i = 0; i < temp.size(); i++){
		cout << temp.at(i);
		if (i == temp.size() - 1)
			break;
		cout << ",";
	}
	cout << "]";

	// cin.get();
	// cin.get();
	return 0;
}

File: C#/338_Counting_Bits.cs
using System;
using System.Collections;

namespace LeetCode
{
    public class Solution
    {
        public int[] CountBits(int num)
        {
            ArrayList ans = new ArrayList();
            ans.Add(0);
            for (int i = 1; i <= num; i++)
            {
                ans.Add((int)ans[(i >> 1)] + (i & 1));
            }
            return (int[])ans.ToArray(typeof(int));
        }

        //Testcase
        static void Main(string[] args)
        {
            Solution s = new Solution();
            Console.Write("Input the int: ");
            int[] temp = s.CountBits(
                Convert.ToInt32(Console.ReadLine()));
            Console.Write("[");
            for (int i = 0; i < temp.Length; i++)
            {
                Console.Write(temp[i]);
                if (i == temp.Length - 1)
                    break;
                Console.Write(",");
            }
            Console.Write("]");
        }
    }
}
